d2/src/main/java/com/linkedin/d2/xds/XdsToD2PropertiesAdaptor.java

```diff
@@ -154,6 +176,31 @@ public void listenToService(String serviceName)
     });
   }
 
+  private boolean shouldNotListenToSymlink(String symlinkName, String fullResourceName)
+  {
+    boolean isSymlink = isSymlinkNode(symlinkName);
+    if (isSymlink)
+    {
+      // use full resource name ("/d2/clusters/$FooClusterMater", "/d2/uris/$FooClusterMaster") as the key
+      // instead of just the symlink name ("$FooClusterMaster") to differentiate clusters and uris symlink resources.
+      _watchedSymlinkResources.computeIfAbsent(fullResourceName, k ->
+      {
+        // use symlink name "$FooClusterMaster" to create the watcher
+        XdsClient.D2SymlinkNodeResourceWatcher watcher = getSymlinkResourceWatcher(symlinkName);
+        _xdsClient.watchXdsResource(fullResourceName, XdsClient.ResourceType.D2_SYMLINK_NODE,
```

Comment:
Nit: better to use `k` rather than `fullResourceName` here since it's inside the mapping function.

Reply:
ah, yeah, why not.